cloudpickle/cloudpickle.py

```diff
@@ -1146,19 +1142,30 @@ def _rehydrate_skeleton_class(skeleton_class, class_dict):
     return skeleton_class
 
 
-def _find_module(mod_name):
+def _can_find_module(mod_name):
     """
     Iterate over each part instead of calling imp.find_module directly.
-    This function is able to find submodules (e.g. scikit.tree)
+    This function is able to find submodules (e.g. os.path).
+
+    Return True if the module can be found.
     """
-    path = None
-    for part in mod_name.split('.'):
-        if path is not None:
-            path = [path]
-        file, path, description = imp.find_module(part, path)
-        if file is not None:
-            file.close()
-    return path, description
+    if hasattr(importlib, 'util'):
+        spec = importlib.util.find_spec(mod_name)
```

Comment:
Note that using this on submodules will import the parent modules.

See the __Warning__ in this [SO answer](https://stackoverflow.com/a/14050282/2642845)

Reply:
Good catch: this is embarrassing as the old method based on imp does not have such a side effect. I don't know what should be done.